src/main/java/tech/allegro/wildsnake/productCategory/model/ProductCategory.java

```diff
@@ -0,0 +1,42 @@
+package tech.allegro.wildsnake.productCategory.model;
+
+import tech.allegro.wildsnake.product.model.Product;
+
+import javax.persistence.*;
+import java.util.HashSet;
+import java.util.Set;
+
+@Entity
+public class ProductCategory {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
```

Comment:
you generate keys in 2 places. One in ProductCategoryDomain and here. Which one is valid? :) I would not recommend share id generated by Mongo with external world.